application/src/main/java/org/togetherjava/tjbot/commands/reminder/RemindRoutine.java

```diff
@@ -0,0 +1,139 @@
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        Instant now = Instant.now();
+        database.write(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.REMIND_AT.lessOrEqual(now))
+            .stream()
+            .forEach(pendingReminder -> {
+                sendReminder(jda, pendingReminder.getId(), pendingReminder.getGuildId(),
+                        pendingReminder.getChannelId(), pendingReminder.getAuthorId(),
+                        pendingReminder.getContent(), pendingReminder.getCreatedAt());
+                pendingReminder.delete();
+            }));
+    }
+
+    private static void sendReminder(@NotNull JDA jda, long id, long guildId, long channelId,
+            long authorId, @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        RestAction<ReminderRoute> route = computeReminderRoute(jda, guildId, channelId, authorId);
+        sendReminderViaRoute(route, id, content, createdAt);
+    }
+
+    private static RestAction<ReminderRoute> computeReminderRoute(@NotNull JDA jda, long guildId,
+            long channelId, long authorId) {
+        // If guild and channel can still be found, send there
+        Guild guild = jda.getGuildById(guildId);
+        if (guild != null) {
+            TextChannel channel = guild.getTextChannelById(channelId);
+            if (channel != null) {
+                return createGuildReminderRoute(jda, authorId, channel);
+            }
+        }
+
+        // Otherwise, attempt to DM the user directly
+        return createDmReminderRoute(jda, authorId);
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createGuildReminderRoute(@NotNull JDA jda,
+            long authorId, @NotNull TextChannel channel) {
+        return jda.retrieveUserById(authorId)
+            .onErrorMap(error -> null)
+            .map(author -> new ReminderRoute(channel, author,
+                    author == null ? null : author.getAsMention()));
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createDmReminderRoute(@NotNull JDA jda,
+            long authorId) {
+        return jda.openPrivateChannelById(authorId)
+            .onErrorMap(error -> null)
+            .map(channel -> channel == null ? new ReminderRoute(null, null, null)
+                    : new ReminderRoute(channel, channel.getUser(),
+                            "(Sending your reminder directly, because I was unable to"
+                                    + " locate the original channel you wanted it to be send to)"));
+    }
+
+    private static void sendReminderViaRoute(@NotNull RestAction<ReminderRoute> routeAction,
+            long id, @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        routeAction.flatMap(route -> {
+            if (route.isUndeliverable()) {
+                throw new IllegalStateException("Route is not deliverable");
+            }
```

Comment:
ReminderRoute#isUndeliverable is always false after the suggestion above, thus remove the method and it's check